client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -514,14 +514,28 @@ class LifecycleManager(appId: String, val conf: CelebornConf) extends RpcEndpoin
       shuffleId: Int,
       oldPartition: PartitionLocation,
       cause: StatusCode): Unit = {
-    // only blacklist if cause is PushDataFailMain
     val failedWorker = new ShuffleFailedWorkers()
-    if (cause == StatusCode.PUSH_DATA_FAIL_MASTER && oldPartition != null) {
-      val tmpWorker = oldPartition.getWorker
-      val worker = workerSnapshots(shuffleId).keySet().asScala
-        .find(_.equals(tmpWorker))
-      if (worker.isDefined) {
-        failedWorker.put(worker.get, (StatusCode.PUSH_DATA_FAIL_MASTER, System.currentTimeMillis()))
+    if (oldPartition != null) {
+      cause match {
+        case StatusCode.PUSH_DATA_FAIL_MASTER =>
```

Comment:
I think it would be better if we extract a local function to avoid duplicate code.

Reply:
> I think it would be better if we extract a local function to avoid duplicate code.

Done